### PR TITLE
4269 refactor fileinputs

### PR DESCRIPTION
**Pull request by Liliya-web**

### CODE REVIEW

**Review by omega790 — COMMENTED** [review]
Before review will be done please check test runs and fix failed tests

**Comment by Liliya-web** [comment]
> Before review will be done please check test runs and fix failed tests

I checked. My tests are not failing. It's some other tests.

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java [hunk 1]
```diff
@@ -158,9 +198,153 @@ public void complexFileInputTest() {
...
+        counterFileInput.has().darkTheme();
+        denseFileInput.show();
+        denseFileInput.has().lightTheme();
+    }
+
+    @Test
+    public void clearIconFileInputTest() {
+        backgroundColorFileInput.show();
+        backgroundColorFileInput.uploadFile(pathTXT.toString());
+        backgroundColorFileInput.is().clearable();
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java:226]: please add:
backgroundColorFileInput.clearButton().click();

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java [hunk 2]
```diff
@@ -158,9 +198,153 @@ public void complexFileInputTest() {
...
+        fullWidthFileInput.show();
+        fullWidthFileInput.has().height(57);
+        fullWidthFileInput.has().heightGreaterThan(50);
+        fullWidthFileInput.has().heightLessThan(60);
+    }
+
+    @Test
+    public void reversedFileInputTest() {
+        loadingFileInput.show();
+        loadingFileInput.is().reversed();
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java:288]: add notReversed()

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java [hunk 3]
```diff
@@ -158,9 +198,153 @@ public void complexFileInputTest() {
...
+    @Test
+    public void placeholderFileInputTest() {
+        selectionFileInput.show();
+        selectionFileInput.has().placeholder("Upload your documents");
+    }
+
+    @Test
+    public void roundedFileInputTest() {
+        prependIconFileInput.show();
+        prependIconFileInput.is().rounded();
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java:314]: add notRounded()

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java [hunk 4]
```diff
@@ -158,9 +198,153 @@ public void complexFileInputTest() {
...
+
+    @Test
+    public void disabledFileInputTest() {
+        disabledFileInput.show();
+        disabledFileInput.is().disabled();
+        soloFileInput.show();
+        soloFileInput.is().enabled();
     }
 }
+
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java:350]: add test for single line

**Review by omega790 — CHANGES_REQUESTED** [review]
Please see the comments

File: jdi-light-vuetify-tests/src/main/java/io/github/com/pages/FileInputsPage.java [hunk 5]
```diff
@@ -14,21 +16,33 @@ public class FileInputsPage extends VuetifyPage {
...
 
     @JDIFileInput(root = "#DenseFileInput.v-file-input")
     public static FileInput denseFileInput;
 
     @JDIFileInput(root = "#MultipleFileInput.v-file-input")
-    public static FileInput multiplyFileInput;
+    public static FileInput multipleFileInput;
 
-    @JDIFileInput(root = "#PrependIconFileInput.v-file-input")
+    @JDIFileInput(root = "//*[@id='AppendPrependIconFileInput']/div[1]")
```
[the remark this answers is not in the document]
Comment (Liliya-web) [reply jdi-light-vuetify-tests/src/main/java/io/github/com/pages/FileInputsPage.java:28]: I think, itćs more convenient to have better names for elements. And I donćt have to get() required element everytime I need it

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java [hunk 6]
```diff
@@ -60,14 +79,16 @@ public void acceptFileInput() {
...
     public void chipsFileInputTest() {
-        chipsFileInput.is().multiply();
+        chipsFileInput.show();
+        chipsFileInput.is().multiple();
         chipsFileInput.uploadFiles(asList(pathTXT.toString(), pathPNG.toString()));
         chipsFileInput.has().files(asList(pathTXT.getFileName().toString(), pathPNG.getFileName().toString()));
     }
 
     @Test
     public void smallChipsFileInputTest() {
```
[the remark this answers is not in the document]
Comment (Liliya-web) [reply jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java:89]: It is not a duplication. First use uploadFiles() and second uploadFile(). I made one test of them

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java [hunk 7]
```diff
@@ -49,19 +49,23 @@ public FileInputAssert disabled() {
 
     @JDIAction("Assert that '{name}' can accept multiple files")
     public FileInputAssert multiple() {
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java:51]: Please use MultipleAssert and delete assert method

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java [hunk 8]
```diff
@@ -87,18 +93,26 @@ public FileInputAssert hasLabel() {
...
                 " to expected suffix text '%s'", actualSuffixText, suffixText));
         return this;
     }
 
     @JDIAction("Assert that '{name}' has prefix text '{0}'")
     public FileInputAssert prefixText(String prefixText) {
         String actualPrefixText = element().prefix().getText();
-        jdiAssert(actualPrefixText, Matchers.equalTo(prefixText), String.format("Actual suffix test '%s' is not equal" +
+        jdiAssert(actualPrefixText, Matchers.equalTo(prefixText), String.format("Actual prefix text '%s' is not equal" +
                 " to expected suffix text '%s'", actualPrefixText, prefixText));
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java:116]: suffix -> prefix

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java [hunk 9]
```diff
@@ -51,4 +92,106 @@ public FileInputAssert hasLabel() {
...
+        return this;
+    }
+
+    @JDIAction("Assert that '{name}' is not full-width")
+    public FileInputAssert notFullWidth() {
+        jdiAssert(element().isFullWidth(), Matchers.is(false), "Element is full-width");
+        return this;
+    }
+
+    @JDIAction("Assert that '{name}' has detailes hidden")
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java:138]: detailes -> details

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java [hunk 10]
```diff
@@ -51,4 +92,106 @@ public FileInputAssert hasLabel() {
...
+
+    @JDIAction("Assert that '{name}' has not details hidden")
+    public FileInputAssert notDetailsHidden() {
+        jdiAssert(element().hasDetailsHidden(), Matchers.is(false), "Element has details hidden");
+        return this;
+    }
+
+    @JDIAction("Assert that '{name}' has loader height {0}")
+    public FileInputAssert loaderHeightPx(int height) {
+        String actualLoaderHeight = element().getLoaderHeight();
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java:152]: please compare numbers as a numbers, not as strings

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/FileInput.java [hunk 11]
```diff
@@ -229,6 +243,70 @@ public FileInput setup(String root, String files) {
...
     }
 
+    @Override
+    public String backgroundColor() {
+        return finds(".v-input__control > .v-input__slot").css("background-color");
+    }
+
+    @Override
+    public String color() {
+        return finds(".v-input__control .v-label").css("color");
```
[the remark this answers is not in the document]
Comment (omega790) [reply jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/FileInput.java:237]: up

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java [hunk 12]
```diff
@@ -140,27 +145,168 @@ public void selectionFileInputTest() {
...
+        disabledFileInput.show();
+        disabledFileInput.is().disabled();
+        soloFileInput.show();
+        soloFileInput.is().enabled();
+    }
+
+    @Test
+    public void singleLineFileInputTest() {
+        prependInnerIconFileInput.show();
+        prependInnerIconFileInput.is().singleLine();
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java:309]: + notSingleLine()

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/interfaces/HasMessages.java [hunk 13]
```diff
@@ -67,4 +85,25 @@ default boolean hasErrorMessages() {
...
         return core().hasClass("success--text");
     }
+
+    /**
+     * Get element's success messages.
+     *
+     * @return list of success messages
+     */
+    @JDIAction("Get '{name}' success messages")
+    default List<String> successMessages() {
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/interfaces/HasMessages.java:95]: Create additional methods with parameters and don't duplicate code

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java [hunk 14]
```diff
@@ -76,62 +84,59 @@ public void smallChipsFileInputTest() {
...
         counterFileInput.has().file("2 files");
         counterFileInput.counter().has().text("2 files (62 B in total)");
     }
 
     @Test
     public void denseFileInputTest() {
-        denseFileInput.uploadFile(pathTXT.toString());
-        denseFileInput.has().file(pathTXT.getFileName().toString());
+        denseFileInput.show();
+        denseFileInput.is().dense();
```
Comment (omega790) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java:101]: + notDense()

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java [hunk 15]
```diff
@@ -147,6 +186,7 @@ public void selectionFileInputTest() {
 
     @Test
     public void complexFileInputTest() {
```
[the remark this answers is not in the document]
Comment (omega790) [reply jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/FileInputsTests.java]: up

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/interfaces/HasMessages.java [hunk 16]
```diff
@@ -22,16 +22,25 @@ default List<UIElement> messages() {
         return core().finds(".v-messages__message");
     }
 
+	/**
+	 * Get element's messages text by locator.
+	 *
+	 * @return list of messages text
+	 */
+	@JDIAction("Get '{name}' messages text by locator")
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/interfaces/HasMessages.java:30]: @JDIAction("Get '{name}' messages text by locator") -> @JDIAction("Get '{name}' messages text by locator '{0}'")

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java [hunk 17]
```diff
@@ -149,9 +143,9 @@ public FileInputAssert notDetailsHidden() {
 
     @JDIAction("Assert that '{name}' has loader height {0}")
     public FileInputAssert loaderHeightPx(int height) {
-        String actualLoaderHeight = element().getLoaderHeight();
-        jdiAssert(actualLoaderHeight, Matchers.equalTo(height + "px"), String.format("Actual element's loader height '%s'" +
-                " is not equal to expected %s", actualLoaderHeight, height + "px"));
+        int actualLoaderHeight = element().getLoaderHeight();
+        jdiAssert(actualLoaderHeight, Matchers.equalTo(height), String.format("Actual element's loader height '%s'" +
+                " is not equal to expected %s", actualLoaderHeight + "px", height + "px"));
```
Comment (omega790) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java:148]: `
String.format("Actual element's loader height '%s' is not equal to expected %s", actualLoaderHeight + "px", height + "px")
->
String.format("Actual element's loader height '%s px' is not equal to expected '%s px'", actualLoaderHeight, height)`

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java (hunk 8 above)
Comment (omega790) [reply jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/FileInputAssert.java:110]: Please update description

**Review by omega790 — CHANGES_REQUESTED** [review]

**Review by omega790 — APPROVED** [review]

**Pull request merged**